### PR TITLE
starting alert docs update

### PR DESCRIPTION
**Pull request by dbanksdesign**
*Issue #, if available:*

*Description of changes:* First update of primitives docs to make use of the new docs. We can use this as a template for updating the other primitives. 

https://user-images.githubusercontent.com/321279/141261368-5630da98-32f5-4be7-9ee1-8451bcec2ecd.mp4

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/src/components/Example.tsx [hunk 1]
```diff
@@ -18,3 +19,32 @@ export function Example({ children, className = '' }: ExampleProps) {
...
+
+export function ExampleCode({ children }) {
+  const [copied, setCopied] = React.useState(false);
+  const [text, setText] = React.useState('');
+  const ref = React.useRef(null);
+
+  React.useLayoutEffect(() => {
+    setText(ref.current.innerText);
+    return () => {};
+  }, [children]);
```
Comment (dbanksdesign) [inline docs/src/components/Example.tsx:31]: Is this super weird/bad code? I wanted to make it easy to add a code block that is passed to prism at the remark/rehype layer and still be able to copy its contents. So I'm adding a ref to a div wrapping the code block and setting its innerText to a state in a layout effect. It works. I don't know if there is a better way to do this...

File: docs/src/pages/components/alert/AlertPropControls.tsx [hunk 2]
```diff
@@ -0,0 +1,96 @@
+import {
```
Comment (dbanksdesign) [inline docs/src/pages/components/alert/AlertPropControls.tsx:1]: Moving components for demo controls to the same directory to clean up the docs/src/components directory so it only contains components for the docs site.

File: docs/src/pages/components/alert/AlertThemeControls.tsx [hunk 3]
```diff
@@ -0,0 +1,107 @@
+import { useState } from 'react';
```
Comment (dbanksdesign) [inline docs/src/pages/components/alert/AlertThemeControls.tsx:1]: Theme controls are still a work in progress...

File: docs/src/pages/components/alert/demo.tsx [hunk 4]
```diff
@@ -1,35 +1,153 @@
...
-import { AlertPropControls } from '@/components/AlertPropControls';
-import { useAlertProps } from '@/components/useAlertProps';
-import { Example } from '@/components/Example';
+import { AlertPropControls } from './AlertPropControls';
+import { useAlertProps } from './useAlertProps';
+import { AlertThemeControls, useAlertThemeProps } from './AlertThemeControls';
+
+const alertTheme = defaultTheme.tokens.components.alert;
+
+const propsToCode = (props) => `import { Alert } from '@aws-amplify/ui-react';
```
Comment (dbanksdesign) [inline docs/src/pages/components/alert/demo.tsx:23]: I don't know if there is a better way to do this. We could use sandpack and have this as an editable example, but would be less performatn.

File: docs/src/pages/components/alert/demo.tsx [hunk 5]
```diff
@@ -1,35 +1,153 @@
...
+              <AlertThemeControls {...alertThemeProps} />
+            </TabItem> */}
+          </Tabs>
+        </Flex>
+        <View style={{ width: '50%', position: 'relative' }}>
+          <Button
+            size="small"
+            className="example-copy-button"
+            disabled={copied}
+            onClick={() => {
```
Comment (dbanksdesign) [inline docs/src/pages/components/alert/demo.tsx:125]: TODO: Update this to use the react copy library.

File: docs/src/pages/components/alert/demo.tsx [hunk 6]
```diff
@@ -1,35 +1,153 @@
...
+              headingLevel={alertProps.headingLevel}
+            >
+              {alertProps.body}
+            </Alert>
+          </AlertThemeProvider>
+          <Tabs>
+            <TabItem title="Props">
+              <AlertPropControls {...alertProps} />
+            </TabItem>
+            {/* <TabItem title="Theme">
```
Comment (dbanksdesign) [inline docs/src/pages/components/alert/demo.tsx:115]: TODO: get theme editor working.

File: docs/src/pages/components/alert/examples/AlertHeadingExample.tsx [hunk 7]
```diff
@@ -0,0 +1,12 @@
+import { Alert } from '@aws-amplify/ui-react';
```
Comment (dbanksdesign) [inline docs/src/pages/components/alert/examples/AlertHeadingExample.tsx:1]: Adding all examples in an examples directory to keep the code shown to uses and code run in sync.

File: package.json [hunk 8]
```diff
@@ -51,7 +51,7 @@
     "@changesets/changelog-github": "0.4.1",
     "husky": ">=6",
     "lint-staged": ">=10",
-    "prettier": "2.3.2",
+    "prettier": "2.4.1",
```
Comment (dbanksdesign) [inline package.json:54]: This update allows prettier to work when there are react components wrapping markdown content like:

```md
<Example>

Testing

</Example>
```

File: docs/src/pages/components/alert/index.page.mdx [hunk 9]
```diff
@@ -1,5 +1,6 @@
 ---
 title: Alert
+description: An Alert displays a brief, important message in a way that attracts the user's attention without interrupting the user's task. Alerts are typically intended to be read out dynamically by a screen reader.
```
Comment (dbanksdesign) [inline docs/src/pages/components/alert/index.page.mdx:3]: Description is put below the title

File: docs/src/pages/components/alert/AlertPropControls.tsx (hunk 2 above)
Comment (reesscot) [reply docs/src/pages/components/alert/AlertPropControls.tsx:1]: Nice, I like this

File: package.json (hunk 8 above)
Comment (reesscot) [reply package.json:54]: 🤔 Out of curiousity, what specific behavior changed between Prettier versions?

Comment (dbanksdesign) [reply package.json:54]: https://prettier.io/blog/2021/09/09/2.4.0.html

It looks like:
* Typescript 4.4 support
* Replace jsxBracketSameLine option with bracketSameLine option

are the main things

File: docs/src/pages/components/alert/AlertThemeControls.tsx (hunk 3 above)
Comment (dbanksdesign) [reply docs/src/pages/components/alert/AlertThemeControls.tsx]: Removed theme controls for now, they are half-baked

File: docs/src/components/Demo.tsx [hunk 10]
```diff
@@ -0,0 +1,53 @@
+import { useState } from 'react';
+import { CopyToClipboard } from 'react-copy-to-clipboard';
+import Highlight, { defaultProps } from 'prism-react-renderer';
+import { Tabs, TabItem, Card, Flex, View, Button } from '@aws-amplify/ui-react';
+
+export const Demo = ({ children, propControls, themeControls, code }) => {
```
Comment (dbanksdesign) [inline docs/src/components/Demo.tsx:6]: This is the component that will be used for primitive demos. It handles the layout and code highlighting and copy button.

File: docs/src/components/Example.tsx [hunk 11]
```diff
@@ -14,7 +15,36 @@ export function Example({ children, className = '' }: ExampleProps) {
       className={`example ${className}`}
       style={{ marginBottom: `${tokens.space.large}` }}
     >
-      {children}
+      <Flex direction="column">{children}</Flex>
```
Comment (dbanksdesign) [inline docs/src/components/Example.tsx:18]: Added this so examples could have multiple elements and be spaced out.

File: docs/src/pages/components/alert/demo.tsx [hunk 12]
```diff
@@ -1,35 +1,47 @@
...
-          hasIcon={alertProps.hasIcon}
-          iconSize={alertProps.iconSize}
-          heading={alertProps.heading}
-          headingLevel={alertProps.headingLevel}
-        >
-          This is the Alert message.
-        </Alert>
-      </Example>
-    </Flex>
+    <Demo
```
Comment (dbanksdesign) [inline docs/src/pages/components/alert/demo.tsx:31]: Demo component takes care of layout and code stuff.

File: docs/src/pages/components/alert/react.mdx [hunk 13]
```diff
@@ -12,113 +19,96 @@ An Alert displays a brief, important message in a way that attracts the user's a
 
 Import the Alert component and styles.
 
-```jsx
-import { Alert } from '@aws-amplify/ui-react';
-import '@aws-amplify/ui-react/styles.css';
+<Example>
```
Comment (dbanksdesign) [inline docs/src/pages/components/alert/react.mdx:22]: How the `Example` and `ExampleCode` components are used.

File: docs/src/components/Demo.tsx (hunk 10 above)
Comment (reesscot) [reply docs/src/components/Demo.tsx:6]: 💯

File: docs/src/components/Demo.tsx [hunk 14]
```diff
@@ -0,0 +1,53 @@
...
+  const [copied, setCopied] = useState(false);
+
+  const copy = () => {
+    setCopied(true);
+    setTimeout(() => {
+      setCopied(false);
```
Comment (reesscot) [inline docs/src/components/Demo.tsx:12]: Love the UX of this. Clear signal that it was copied, but still lets you copy again if you want later on

File: docs/src/pages/components/alert/demo.tsx (hunk 12 above)
Comment (reesscot) [reply docs/src/pages/components/alert/demo.tsx:31]: Sweeet

**Review by reesscot — APPROVED** [review]
This looks so clean, thanks for refactoring!

**Pull request merged**